integration-tests/src/test/java/com/amazonaws/services/schemaregistry/integrationtests/mm2converter/GlueSchemaRegistryConverterIntegrationTest.java

```diff
@@ -0,0 +1,163 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class GlueSchemaRegistryConverterIntegrationTest {
+
+    private static CrossRegionReplicationMM2Converter converter;
+
+    private static List<String> schemasToCleanUp = new ArrayList<>();
+
+    private static final AwsCredentialsProvider defaultCredProvider = DefaultCredentialsProvider
+            .builder()
+            .build();
+
+    @BeforeEach
+    public void setUp() {
+        converter = new CrossRegionReplicationMM2Converter();
+    }
+
+    @AfterAll
+    public static void tearDown() throws URISyntaxException {
+        Map<String, Object> properties = getConverterProperties();
+        GlueClient glueClientAtSourceRegion = GlueClient.builder()
+                .credentialsProvider(defaultCredProvider)
+                .region(Region.of((String) properties.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION)))
+                .endpointOverride(new URI((String) properties.get(AWSSchemaRegistryConstants.AWS_SOURCE_ENDPOINT)))
+                .httpClient(UrlConnectionHttpClient.builder()
+                        .build())
+                .build();
+
+        GlueClient glueClientAtTargetRegion = GlueClient.builder()
+                .credentialsProvider(defaultCredProvider)
+                .region(Region.of((String) properties.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION)))
+                .endpointOverride(new URI((String) properties.get(AWSSchemaRegistryConstants.AWS_TARGET_ENDPOINT)))
+                .httpClient(UrlConnectionHttpClient.builder()
+                        .build())
+                .build();
+
+        for (String schemaName : schemasToCleanUp) {
+            DeleteSchemaRequest deleteSchemaRequest = DeleteSchemaRequest.builder()
+                    .schemaId(SchemaId.builder()
+                            .registryName("default-registry")
+                            .schemaName(schemaName)
+                            .build())
+                    .build();
+
+            glueClientAtSourceRegion.deleteSchema(deleteSchemaRequest);
+            glueClientAtTargetRegion.deleteSchema(deleteSchemaRequest);
+        }
+    }
+
+    @Test
+    public void SchemaReplication_AvroData_Succeeds() {
+        converter.configure(getConverterProperties(), false);
+        AWSKafkaAvroSerializer avroSerializer = new AWSKafkaAvroSerializer(defaultCredProvider, getSourceGSRProperties());
+
+        List<Schema> schemaList = AvroSchemaGenerator.createTestSchema();
+        List<GenericRecord> recordList = AvroSchemaGenerator.createTestRecord();
+        int index = 0;
+        for (org.apache.avro.Schema schema : schemaList){
+            GenericRecord record = recordList.get(index);
+            byte[] serialize = avroSerializer.serialize(schema.getName(), record);
+
+            com.amazonaws.services.schemaregistry.common.Schema returnSchema = converter.getSchema(serialize);
```

Comment:
this is not the integration test I am looking for, what I was expecting from your integration test is you integrate your converter with actual mirror maker and 2 kafka clusters. Then in your integration test you should produce to source cluster and then consume from target cluster to verify the data integrity. Then you need to verify the schema integrity by comparing the schema in source region and target region GSR